pennylane_lightning/lightning_qubit.py

```diff
@@ -173,7 +173,18 @@ class LightningQubit(QubitDevice):
     operations = allowed_operations
     observables = allowed_observables
 
-    def __init__(self, wires, *, c_dtype=np.complex128, shots=None, batch_obs=False, analytic=None):
+    def __init__(
+        self,
+        wires,
+        *,
+        c_dtype=np.complex128,
+        shots=None,
+        mcmc=False,
+        kernel_name="Local",
+        num_burnin=100,
```

Comment:
We may need to document these somewhere, otherwise it will be lost into the void. Should the device documentation suffice for these, while we aren't expecting them to be a full-on multi-device implementation? Tagging @trbromley  for thoughts here.

Reply:
Hi @trbromley, how do you think?

Reply:
Yes would be good to mention the new arguments here: https://docs.pennylane.ai/projects/lightning/en/stable/code/api/pennylane_lightning.LightningQubit.html

and potentially at the bottom of here: https://docs.pennylane.ai/projects/lightning/en/stable/devices.html
Similar to how we have some additional info on parallel adjoint.

Reply:
So if `mcmc=False` then the values for `kernel_name` and `num_burnin` are ignored? I wonder if there's a way to have fewer than 3 arguments.

Reply:
Here is the timing benchmark results from the python layer ![bench_ghz_10000_True_True_mcmc](https://user-images.githubusercontent.com/31480676/225330496-627158d4-c3b9-411b-a945-b6257099e1da.png)
![bench_ghz_100000_True_True_mcmc](https://user-images.githubusercontent.com/31480676/225330641-4c12df9c-220a-45e2-aa3f-40125a07b92c.png)

Reply:
Here are the accuracy comparison:
 
![bench_ghz_20_False_False_mcmc](https://user-images.githubusercontent.com/31480676/225331363-f391e531-8424-4744-8b03-d1142ec727f4.png)
![bench_ghz_25_False_False_mcmc](https://user-images.githubusercontent.com/31480676/225331446-e331ac34-bce8-4cff-b81b-d3dee5ed94ab.png)

Reply:
Thanks Shuli --- for the relative error plots, are these averaged over multiple runs, or a single run?
They look very noisy, so we should ideally try this a few times and see if the results are consistent.

Reply:
Hi @mlxd, It's a single run for multiple shots. Let me run many times and update it then.

Reply:
> So if `mcmc=False` then the values for `kernel_name` and `num_burnin` are ignored? I wonder if there's a way to have fewer than 3 arguments.

Hi @trbromley, per few arguments when loading a device , for sure, we can reduce the number of arguments at `ctor` by moving the `kernel_name` and `num_burnin` to the method `generate_samples(kernel_name, num_burnin)`. But I would like to suggest to put these arguments at the `ctor` since it will raise error if `kernel_name` and `num_burnin` are not correctly set before running circuit. What do you think?

Reply:
> Hi @trbromley, per few arguments when loading a device , one way that we can reduce the number of arguments at ctor is to move the kernel_name and num_burnin to the method generate_samples(kernel_name, num_burnin). But I would like to suggest to put these arguments at the ctor since it will raise error if kernel_name and num_burnin are not correctly set before running circuit. What do you think?

Ok sounds good, happy to keep the current approach :+1:

Reply:
I'm happy to resolve this issue now